### PR TITLE
Routines to validate input and output parameters for Direct reduction

### PR DESCRIPTION
**Pull request by abuts**
Jenkins, test this please

### CODE REVIEW

**Comment by abuts** [comment]
completed as described at 
http://trac.mantidproject.org/mantid/ticket/11063#comment:12

**Comment by martyngigg** [comment]
Could you provide a better title for the pull request? You can edit it with the *Edit* button on the top right of the page

**Comment by abuts** [comment]
Jenkins, test this please

**Comment by DanNixon** [comment]
Jenkins, retest this please.

File: Code/Mantid/scripts/Inelastic/Direct/PropertiesDescriptors.py [hunk 1]
```diff
@@ -307,8 +313,22 @@ def next(self): # Python 3: def __next__(self)
...
+       inc_en = self._incident_energy
+       if isinstance(inc_en,list):
+           for ind,en in enumerate(inc_en):
+             if en <= 0:
+               return (False,2,"Incident energy have to be positive number or list of positive numbers.\n" +
+                               "For input argument {0} got negative energy {1}".format(ind,en))
+       else:
+         if inc_en <= 0:
+            return (False,2,"Incident energy have to be positive number or list of positive numbers.\n" +
+                             "Got single negative incident energy {0} ".format(en))
```
Comment (DanNixon) [inline Code/Mantid/scripts/Inelastic/Direct/PropertiesDescriptors.py]: Should the ``en``` on this line be ```inc_en```?
Setting an incident energy of zero or less gives an UnboundLocalError here.

**Comment by DanNixon** [comment]
I also seem to always get this error from the Direct Convert to Energy UI:
```
UnboundLocalError: local variable 'result' referenced before assignment
  at line 20 in '<Interface>'
  caused by line 519 in '/home/dan/testing-builds/test1/Code/Mantid/scripts/Inelastic/Direct/DirectEnergyConversion.py'
```

**Comment by abuts** [comment]
I am aware GUI may be broken by the changes in direct reduction scripts, but this ticket is one of the series of changes and the gui was broken by one of the previous changes/ To separate changes in C++ from changes in python and not to carry changing in C++ with every change in python, I've created separate ticket #11155, dependent on all python changes completion. GUI should be fixed there.

**Comment by DanNixon** [comment]
There now seems to be some issues with indentation, when I try to run ```MariReduction.py``` I get the following error:
```
IndentationError: unexpected indent (PropertyManager.py, line 162)
  at line 3 in '/home/dan/testing-builds/test1/Code/Mantid/scripts/test/MariReduction.py'
  caused by line 6 in '/home/dan/testing-builds/test1/Code/Mantid/scripts/Inelastic/Direct/ReductionWrapper.py'
```

**Comment by DanNixon** [comment]
Jenkins, retest this please.

**Comment by abuts** [comment]
Jenkins, retest this please

**Comment by abuts** [comment]
Jenkins, retest this please.

**Comment by abuts** [comment]
Jenkins, retest this please

**Comment by abuts** [comment]
Jenkins, reyest this please

**Comment by abuts** [comment]
Jenkins, retest this please

**Comment by abuts** [comment]
Jenkins, retest this please

**Comment by abuts** [comment]
Jenkins, retest this please

**Comment by abuts** [comment]
Jenkins, retest this please

**Comment by martyngigg** [comment]
Jenkins, retest this please.

**Comment by abuts** [comment]
Build seems goes constantly wrong on this one, despite nothing looks wrong with code. Closing trying to bypass git problem

**Pull request closed without merging**